### PR TITLE
[Rated Disabilities] Integrate new LH endpoint behind feature flag

### PR DESCRIPTION
**Pull request by jerekshoe**
## Summary
Adding versions of the UI components to be used with Lighthouse behind the `rd_use_lighthouse` feature toggle

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#63310

### vets-api
**Note:** Make sure to pull the latest `vets-api-mockdata` and `vets-api`
1) Open `settings.local.yml`
2) Add the below:
```yaml
lighthouse:
  veteran_verification:
    host: https://staging-api.va.gov
    form526:
      host: https://staging-api.va.gov
    use_mocks: true
```

### vets-website
**Note:** Make sure that the `rated_disabilities_use_lighthouse` feature flag is enabled
1) Log in using mocked auth [here](http://localhost:3001/sign-in/mocked-auth)
* Credential Service Provider: ID.me
* Profile: vets.gov.user+228@gmail.com
2) Navigate to [http://localhost:3001/disability/view-disability-rating/rating](http://localhost:3001/disability/view-disability-rating/rating)

## Testing done
Added specs for new components

## Screenshots
Screenshot 2024-02-26 at 3 42 23 PM

## What areas of the site does it impact?
Rated Disabilities Application

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/rated-disabilities/components/RatedDisabilityView.jsx [hunk 1]
```diff
@@ -4,41 +4,13 @@ import { CONTACTS } from '@department-of-veterans-affairs/component-library/cont
...
 import { checkForDiscrepancies } from '../actions';
+import MVIError from './MVIError';
 import TotalRatedDisabilities from './TotalRatedDisabilities';
 import OnThisPage from './OnThisPage';
-import RatedDisabilityList from './RatedDisabilityList';
+import RatedDisabilityList from './evss/RatedDisabilityList';
 
 const facilityLocatorUrl = getAppUrl('facilities');
 
-const renderMVIError = () => {
```
Comment (pmclaren19) [inline src/applications/rated-disabilities/components/RatedDisabilityView.jsx:13]: Nice much cleaner!

File: src/applications/rated-disabilities/actions/index.js [hunk 2]
```diff
@@ -111,6 +111,74 @@ export function fetchTotalDisabilityRating(recordAnalyticsEvent = recordEvent) {
   };
 }
 
+export function getRatedDisabilities() {
+  return new Promise(resolve => {
```
Comment (jerekshoe) [inline src/applications/rated-disabilities/actions/index.js:115]: Need to hook up the actual all to vets-api

File: src/applications/rated-disabilities/components/CombinedRating.jsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import recordEvent from 'platform/monitoring/record-event';
```
Comment (jerekshoe) [inline src/applications/rated-disabilities/components/CombinedRating.jsx:4]: Fix this import

File: src/applications/rated-disabilities/components/FeatureFlagsLoaded.jsx [hunk 4]
```diff
@@ -10,11 +10,11 @@ const loadingIndicator = (
   </div>
 );
 
-export default function AppContent({ children, featureFlagsLoading }) {
+export default function FeatureFlagsLoaded({ children, featureFlagsLoading }) {
```
Comment (jerekshoe) [inline src/applications/rated-disabilities/components/FeatureFlagsLoaded.jsx:13]: Pull this out into a separate PR

File: src/applications/rated-disabilities/components/MVIError.jsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+
+import { getAppUrl } from '@department-of-veterans-affairs/platform-utilities/exports';
+
+const facilityLocatorUrl = getAppUrl('facilities');
+
+export default function MVIError() {
```
Comment (jerekshoe) [inline src/applications/rated-disabilities/components/MVIError.jsx:7]: Need to talk to Robin later and determine if this is actually needed or if I can just rely on the `verify` prop on the `RequiredLoginView` component

File: src/applications/rated-disabilities/components/RatingList/ListItem.jsx [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+  const headingParts = [diagnosticText];
+  if (ratingPercentage !== null) {
+    headingParts.unshift(`${ratingPercentage}% rating for`);
+  }
+
+  return headingParts.join(' ');
+};
+
+const RatedDisabilityListItem = ({ ratedDisability }) => {
```
Comment (jerekshoe) [inline src/applications/rated-disabilities/components/RatingList/ListItem.jsx:19]: Rename this to just `ListItem` to match the filename

File: src/applications/rated-disabilities/components/RatingList/RatingList.jsx [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+  getNonServiceConnectedRatings,
+} from './helpers';
+import NoRatings from './NoRatings';
+import Section from './Section';
+
+export default function RatingList({ ratings }) {
+  const serviceConnectedRatings = getServiceConnectedRatings(ratings);
+  const nonServiceConnectedRatings = getNonServiceConnectedRatings(ratings);
+
+  // NOTE: These are true for now to mimic the behavior of the EVSS implementation. It probably makes
```
Comment (jerekshoe) [inline src/applications/rated-disabilities/components/RatingList/RatingList.jsx:15]: Move this comment, it's not relevant to `hasRatings`

File: src/applications/rated-disabilities/components/RatingList/Section.jsx [hunk 8]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import ListItem from './ListItem';
+
+export default function Section({ ratings }) {
```
Comment (jerekshoe) [inline src/applications/rated-disabilities/components/RatingList/Section.jsx:6]: Maybe rename this to `List` and rename `RatingList` to `RatingLists` because it makes more sense

File: src/applications/rated-disabilities/components/RatingList/index.jsx [hunk 9]
```diff
@@ -0,0 +1,3 @@
+import RatingList from './RatingList';
```
Comment (jerekshoe) [inline src/applications/rated-disabilities/components/RatingList/index.jsx:1]: I could have included the `RatingList` component code in here, but I wanted to have the `RatingList` code be in a file named `RatingList.jsx` so it's easier to find

File: src/applications/rated-disabilities/components/TotalRatedDisabilities.jsx [hunk 10]
```diff
@@ -40,3 +40,21 @@ TotalRatedDisabilities.propTypes = {
 };
 
 export default TotalRatedDisabilities;
+
+// content = (
```
Comment (jerekshoe) [inline src/applications/rated-disabilities/components/TotalRatedDisabilities.jsx:44]: Need to determine where this needs to go. Most likely this is the content that should appear when there is a server error on the API call

File: src/applications/rated-disabilities/components/evss/RatedDisabilityList.jsx [hunk 11]
```diff
@@ -3,9 +3,9 @@ import { CONTACTS } from '@department-of-veterans-affairs/component-library/cont
 import PropTypes from 'prop-types';
```
Comment (jerekshoe) [inline src/applications/rated-disabilities/components/evss/RatedDisabilityList.jsx:3]: This doesn't need to go in a separate folder

File: src/applications/rated-disabilities/containers/App.jsx [hunk 12]
```diff
@@ -9,12 +9,14 @@ import { RequiredLoginView } from '@department-of-veterans-affairs/platform-user
 import backendServices from '@department-of-veterans-affairs/platform-user/profile/backendServices';
```
Comment (jerekshoe) [inline src/applications/rated-disabilities/containers/App.jsx:9]: This doesn't need to go in a separate folder

File: src/applications/rated-disabilities/containers/App.jsx [hunk 13]
```diff
@@ -24,6 +26,7 @@ const App = props => {
     <RequiredLoginView
       serviceRequired={backendServices.USER_PROFILE}
       user={props.user}
+      verify
```
Comment (jerekshoe) [inline src/applications/rated-disabilities/containers/App.jsx:29]: Need to talk to Robin about this, will do that today

File: src/applications/rated-disabilities/containers/AppContent.jsx [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+const loadingIndicator = (
+  <va-loading-indicator message="Loading your rating information..." />
+);
+
+export default function AppContent() {
+  const [data, setData] = useState({});
+  const [dataLoaded, setDataLoaded] = useState(false);
+
+  useEffect(() => {
+    getRatedDisabilities().then(res => {
```
Comment (jerekshoe) [inline src/applications/rated-disabilities/containers/AppContent.jsx:17]: Need to add error handling here

File: src/applications/rated-disabilities/containers/AppContent.jsx [hunk 15]
```diff
@@ -0,0 +1,55 @@
...
+      <div className="vads-l-row">
+        <div className="vads-l-col--12">
+          <h1>View your VA disability ratings</h1>
+          {dataLoaded ? (
+            <>
+              <div>{hasRatedDisabilities && <OnThisPage />}</div>
+              <h2 id="combined-rating" className="vads-u-margin-y--1p5">
+                Your combined disability rating
+              </h2>
+              <TotalRatedDisabilities
```
Comment (jerekshoe) [inline src/applications/rated-disabilities/containers/AppContent.jsx:38]: This should be the `CombinedRating` component

File: src/applications/rated-disabilities/containers/AppContent.jsx [hunk 16]
```diff
@@ -0,0 +1,55 @@
...
+                totalDisabilityRating={combinedDisabilityRating}
+                loading={false}
+                error={null}
+              />
+              <h2 id="individual-ratings" className="vads-u-margin-y--2">
+                Your individual ratings
+              </h2>
+              <RatingList ratings={individualRatings} />
```
Comment (jerekshoe) [inline src/applications/rated-disabilities/containers/AppContent.jsx:46]: Rename this to `RatingsList`

File: src/applications/rated-disabilities/components/CombinedRating.jsx [hunk 17]
```diff
@@ -0,0 +1,39 @@
...
+
+import { missingTotalMessage } from './TotalRatingStates';
+
+export default function CombinedRating({ combinedRating }) {
+  if (!combinedRating) {
+    return missingTotalMessage();
```
[the remark this answers is not in the document]
Comment (pmclaren19) [reply src/applications/rated-disabilities/components/CombinedRating.jsx:10]: Agree! Especially since this is in the components folder already.

File: src/applications/rated-disabilities/components/CombinedRating.jsx [hunk 18]
```diff
@@ -0,0 +1,39 @@
...
+
+  return (
+    <va-featured-content>
+      <h3 slot="headline">{heading}</h3>
+      <p>
+        This rating doesn’t include any conditions from claims that we’re still
+        reviewing. You can check the status of your disability claims, decision
+        reviews, or appeals online.
+      </p>
+      <a
```
[the remark this answers is not in the document]
Comment (pmclaren19) [reply src/applications/rated-disabilities/components/CombinedRating.jsx:23]: yea looks like it has the same props...doesnt look like there is a difference so id just change it.

[the remark this answers is not in the document]
Comment (jerekshoe) [reply src/applications/rated-disabilities/components/CombinedRating.jsx:23]: I just need to make sure the `onClick` will still work, may need to import the react bindings to get it working

**Review by pmclaren19 — CHANGES_REQUESTED** [review]

File: src/applications/rated-disabilities/components/Learn.jsx [hunk 19]
```diff
@@ -18,7 +18,6 @@ export default function Learn() {
       <va-link
         href="/disability/about-disability-ratings"
         text="About VA disability ratings"
-        uswds="false"
```
Comment (pmclaren19) [inline src/applications/rated-disabilities/components/Learn.jsx:21]: There is a ticket/pr for this work so I wouldnt update here its out of scope

File: src/applications/rated-disabilities/components/NeedHelp.jsx [hunk 20]
```diff
@@ -3,14 +3,13 @@ import { CONTACTS } from '@department-of-veterans-affairs/component-library/cont
 
 export default function NeedHelp() {
   return (
-    <va-need-help class="vads-u-margin-y--3" uswds="false">
```
Comment (pmclaren19) [inline src/applications/rated-disabilities/components/NeedHelp.jsx:6]: There is a ticket/pr for this work so I wouldnt update here its out of scope

File: src/applications/rated-disabilities/components/RatingLists/RatingLists.jsx [hunk 21]
```diff
@@ -0,0 +1,58 @@
...
+  const hasServiceConnectedRatings = true;
+  const hasNonServiceConnectedRatings = true;
+
+  if (!hasRatings) {
+    return <NoRatings />;
+  }
+
+  return (
+    <>
+      {hasServiceConnectedRatings && (
```
[the remark this answers is not in the document]
Comment (jerekshoe) [reply src/applications/rated-disabilities/components/RatingLists/RatingLists.jsx:25]: Yes, and it's possible that neither of them are displayed as well

File: src/applications/rated-disabilities/components/NeedHelp.jsx [hunk 22]
```diff
@@ -3,14 +3,13 @@ import { CONTACTS } from '@department-of-veterans-affairs/component-library/cont
...
+    <va-need-help class="vads-u-margin-y--3">
       <div slot="content">
         <p>
-          You can call us at{' '}
-          <va-telephone contact={CONTACTS.VA_BENEFITS} uswds="false" />. We’re
-          We’re here Monday through Friday, 8:00 a.m to 9:00 p.m. ET. If you
-          have hearing loss, call{' '}
-          <va-telephone contact={CONTACTS['711']} tty uswds="false" />
```
Comment (pmclaren19) [inline src/applications/rated-disabilities/components/NeedHelp.jsx:13]: There is a ticket/pr for this uswds work so I wouldnt update here its out of scope

File: src/applications/rated-disabilities/components/NeedHelp.jsx [hunk 23]
```diff
@@ -3,14 +3,13 @@ import { CONTACTS } from '@department-of-veterans-affairs/component-library/cont
 
 export default function NeedHelp() {
   return (
-    <va-need-help class="vads-u-margin-y--3" uswds="false">
+    <va-need-help class="vads-u-margin-y--3">
       <div slot="content">
         <p>
-          You can call us at{' '}
-          <va-telephone contact={CONTACTS.VA_BENEFITS} uswds="false" />. We’re
```
Comment (pmclaren19) [inline src/applications/rated-disabilities/components/NeedHelp.jsx:10]: There is a ticket/pr for this uswds work so I wouldnt update here its out of scope

File: src/applications/rated-disabilities/components/NoCombinedRating.jsx [hunk 24]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+
+export default function NoCombinedRating() {
+  return (
+    <va-alert status="info" uswds>
```
Comment (pmclaren19) [inline src/applications/rated-disabilities/components/NoCombinedRating.jsx:5]: if we are adding a new va-alert to rated disabilities that has uswds enabled then we will want to update the ticket [here](https://github.com/department-of-veterans-affairs/va.gov-team/issues/76124) and spreadsheet [here](https://docs.google.com/spreadsheets/d/1ajKjN6dr-2PzpgEbfX1OqxAupvbqoHwqh7ml1m5a1Fo/edit?usp=sharing) to make sure we account for this work.

File: src/applications/rated-disabilities/components/RatingLists/NoRatings.jsx [hunk 25]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+
+export default function NoRatings() {
+  return (
+    <va-alert uswds>
```
Comment (pmclaren19) [inline src/applications/rated-disabilities/components/RatingLists/NoRatings.jsx:5]: if we are adding a new va-alert to rated disabilities that has uswds enabled then we will want to update the ticket https://github.com/department-of-veterans-affairs/va.gov-team/issues/76124 and spreadsheet [here](https://docs.google.com/spreadsheets/d/1ajKjN6dr-2PzpgEbfX1OqxAupvbqoHwqh7ml1m5a1Fo/edit?usp=sharing) to make sure we account for this work.

File: src/applications/rated-disabilities/components/ServerError.jsx [hunk 26]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { CONTACTS } from '@department-of-veterans-affairs/component-library/contacts';
+
+export default function ServerError() {
+  return (
+    <va-alert status="error" uswds>
```
Comment (pmclaren19) [inline src/applications/rated-disabilities/components/ServerError.jsx:6]: if we are adding a new va-alert to rated disabilities that has uswds enabled then we will want to update the ticket https://github.com/department-of-veterans-affairs/va.gov-team/issues/76124 and spreadsheet [here](https://docs.google.com/spreadsheets/d/1ajKjN6dr-2PzpgEbfX1OqxAupvbqoHwqh7ml1m5a1Fo/edit?usp=sharing) to make sure we account for this work.

File: src/applications/rated-disabilities/tests/e2e/rated-disabilities-lighthouse.cypress.spec.js [hunk 27]
```diff
@@ -0,0 +1,123 @@
...
+//     statusCode: 404,
+//   }).as('totalRatingClientError');
+
+//   cy.findByText(
+//     /We don’t have a combined disability rating on file for you/,
+//   ).should('exist');
+//   cy.findByText(/We don’t have rated disabilities on file for you/).should(
+//     'exist',
+//   );
+// };
```
Comment (samcoforma) [inline src/applications/rated-disabilities/tests/e2e/rated-disabilities-lighthouse.cypress.spec.js:37]: Do we still need these commented tests?

File: src/applications/rated-disabilities/components/NoCombinedRating.jsx (hunk 24 above)
Comment (jerekshoe) [reply src/applications/rated-disabilities/components/NoCombinedRating.jsx:5]: Added to spreadsheet

File: src/applications/rated-disabilities/components/RatingLists/NoRatings.jsx (hunk 25 above)
Comment (jerekshoe) [reply src/applications/rated-disabilities/components/RatingLists/NoRatings.jsx:5]: Added to spreadsheet

File: src/applications/rated-disabilities/components/ServerError.jsx (hunk 26 above)
Comment (jerekshoe) [reply src/applications/rated-disabilities/components/ServerError.jsx:6]: Added to spreadsheet

**Review by samcoforma — APPROVED** [review]

**Review by pmclaren19 — APPROVED** [review]
